provisioning/utils.go

```diff
@@ -47,8 +48,27 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetIronicIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec) (string, error) {
-	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
+		return "", err
+	}
+
+	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
+	if infra.Status.PlatformStatus.BareMetal == nil {
+		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+		return "", err
+	}
+
+	// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
```

Comment:
Can be a follow-up as well IMO to avoid updating vendored dependencies in the same change.

Reply:
:+1: